### DISCUSSION TITLE
ESP32-C3 support

### DISCUSSION THREAD

**Discussion by Avamander** [bxparks/AceSPI#5] [General]
I really want to use AceSegment with my ESP32-C3-DevKitM-1 board, but whatever configuration I try for the SPI bus, I can't get it to function. (For example https://www.esp32.com/viewtopic.php?t=20389&start=10 or the configuration mentioned here https://github.com/BugerDread/cryptoticker-esp32-c3)

I (and maybe others) would really appreciate if you could provide an example that should in theory to work, that would also allow me to attach a logic analyzer and see what's going on (if it doesn't work).

**Comment by bxparks** [comment]
I don't have an ESP32C3 so can't help very much. Did you try using a normal ESP32 or one of the other supported microcontrollers to get your program working? Then transferring the code to the C3? Also try starting with the simplest and most minimal code as possible, get that working. Then build up from there.

**Reply by Avamander** [reply]
I used AceSegment's own MAX7219 example unmodified initially, but then with various pin mappings.

**Reply by bxparks** [reply]
You mean the `AceSegment/examples/HelloMax7219` code? That one should work on the boards that I've tested. But I don't have the C3, so I don't know about that one.

**Reply by Avamander** [reply]
> You mean the AceSegment/examples/HelloMax7219 code?

Yep.

> That one should work on the boards that I've tested. But I don't have the C3, so I don't know about that one.

Thought so as well. I just suspect that some pin somewhere is wrong, but it's not clear at all which and where.

**Reply by bxparks** [reply]
I double-checked the `examples/HelloMax7219` code on my ESP32, in case I had a typo or something, and it works just fine on the ESP32. I checked on a SparkFun Pro Micro as well, also works fine. The ESP32 has 2 SPI buses, and my dev board was wired up to connect the MAX7219 to the secondary SPI bus, so the pins had to be remapped, and an explicit secondary instance of `SPIClass` had to created, but other than that, everything worked.
